client/src/types/index.tsx

```diff
@@ -15,8 +16,14 @@ export type FullResponse<T> = AxiosResponse<ApiResponse<T>, unknown>;
 export type Numeric = number | string;
 
 export interface Message {
-  msg: string,
+  msg: string | Array<string>,
   severity?: 'error' | 'warning' | 'info' | 'success'
 }
 
 export type State<T> = [T, (value: T) => void];
+
+export interface NewCourseData extends Omit<
+CourseData, 'teachersInCharge' | 'id' | 'evaluationInformation'
+> {
+  teachersInCharge: Array<string>;
+}
```

Comment:
I think it would be cleaner if we avoid adding new types for POST request data unless it is somehow significantly different from the corresponding GET request data. In this case, I would add an optional `email` field to `UserData` and make `evaluationInformation` optional (or completely remove it since it has remained unused ever since it was added) in `CourseData`, and then we could just use that.

Reply:
Good point, remove the new type and updated old one. Kept `evaluationInformation` as is (set as optional).